src/file/source/file.rs

```diff
@@ -59,9 +59,10 @@ impl FileSourceFile {
                 )))
             };
         }
-        // Adding a dummy extension will make sure we will not override secondary extensions, i.e. "file.local"
-        // This will make the following set_extension function calls to append the extension.
-        let mut filename = add_dummy_extension(filename);
+
+        // Append an extension that will be replaced when calling `set_extension()`:
```

Comment:
Is this a better revision?

```suggestion
        // Add a fake extension which will be replaced by `set_extension()` calls.
        // Prevents accidentally replacing a portion of the base filename with an existing `.` (ref: #306)
```